src/include/dng/hts/bcf.h

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2015 Reed A. Cartwright
+ * Authors:  kdai1 <kdai1@asu.edu>
+ *
+ * This file is part of DeNovoGear.
+ *
+ * DeNovoGear is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+#ifndef CXX_HTS_BCF_H
+#define CXX_HTS_BCF_H
+
+#include "hts.h"
+
+#include <vector>
+#include <string>
+#include <set>
+#include <htslib/vcf.h>
+
+
+#define VCF_VERSION "VCFv4.1"
+
+namespace hts { namespace bcf { 
+
+/**
+ * File - Class for writing VCF/BCF files to the output, either stdout or to a file
+ * To write to stdout set file="-" in constructor. To write as a BCF file set mode="wb" in constructor.
+ * To use:
+ *  1. call addHeaderMetadata() and addSample() to add information to the VCF header
+ *  2. call WriteHdr() to finish the header
+ *  3. populate the first few columns of a VCF body line using SetID(), SetFilter(), SetAlleles(), UpdateInfo() and UpdateSamples() 
+ *  4. call WriteRecord() to save/print the current VCF line and move on to the next
+ */
+class File : hts::File
+{
+ public:
+  
+  /**
+   * Initialize File/output stream for writing VCF/BCF
+   * @file - file name to write to, use "-" for stdout
+   * @mode: "w" = write VCF, "wb" = write BCF
+   * @source: name of application writing the VCF file
+   */
+  File(const char *file, const char *mode, const char *source)
+    : hts::File(file, mode)
+    {
+      hdr = bcf_hdr_init("w"); // VCF header
+      rec = bcf_init1(); // VCF record/body line
+      
+      std::string s = std::string("#source=") + source;
+      bcf_hdr_append(hdr, s.c_str());
+    }
+
+  File(const File&) = delete;
+
+  File& operator=(File& other)
```

Comment:
We need a move operator, not a copy one.  See https://github.com/denovogear/denovogear/blob/feature/newcaller/src/include/dng/hts/bam.h#L65  Note the use of an rvalue reference.